### PR TITLE
ATM Endpoint filtering improvements

### PR DESCRIPTION
**Pull request by esbena**
Based on sheet `atm-results (2021-12-08)`: 

- 66 triaged results
- 45 addressed FP results: should disappear

- Primary approach: extend the endpoint filters directly
- Secondary approach: model more database accesses

The secondary approach requires some manual result inspection, especially for the missing rate-limiting query.

Core library changes:

- More database accesses recognized (redis, ioredis, marsdb)
- The `express-fileupload` `mv` function has a path-injection sink

### CODE REVIEW

File: javascript/ql/lib/semmle/javascript/frameworks/NoSQL.qll
Comment (asgerf) [inline javascript/ql/lib/semmle/javascript/frameworks/NoSQL.qll:629]: On these new `DatabaseAccess` subclasses (this one and the others), there is no restriction on the method name. Have you checked the API that this is reasonable? Maybe for the ATM use-case it doesn't matter, but there may be other uses where you don't want calls that aren't actually accessing the database.

For example, I'm not sure registering an event handler ought to be considered a database access.

Comment (asgerf) [reply javascript/ql/lib/semmle/javascript/frameworks/NoSQL.qll:629]: Btw this isn't a blocker if merging this PR would unblock others.

Comment (esbena) [reply javascript/ql/lib/semmle/javascript/frameworks/NoSQL.qll:629]: I did think along those lines, but I think all siblings to methods that could run queries are fair game in practice. For your example and the missing rate-limit query, I suppose it could be real problem if you add lots of event handlers to an active database connection.

The evaluation did not show any new alerts from this change.

Comment (esbena) [reply javascript/ql/lib/semmle/javascript/frameworks/NoSQL.qll:629]: So I'll merge as is, and volunteer to fix the FPs this heuristic causes.

**Comment by esbena** [comment]
Had to format, I need a new approval.

**Review by Z80coder — APPROVED** [review]
Thanks @esbena and @asgerf!

**Pull request merged**
